index.d.ts

```diff
@@ -0,0 +1,758 @@
+   * @param {object?} ths this argument
+   * @returns {boolean} true if subsequence
+   */
+  export function isSubsequenceOn<T>(x: T[], y: T[], fn?: (v: T, i: number, x: T[]) => T[], ths?: object): boolean;
+
+  /**
+   * Checks if array ends with a suffix.
+   * @param {Array} x an array
+   * @param {Array} y suffix?
+   * @param {function?} fn compare function (a, b)
+   * @returns {boolean} true if suffix
+   */
+  export function isSuffix<T>(x: T[], y: T[], fn?: (a: T, b: T) => boolean): boolean;
+
+  /**
+   * Checks if array ends with a suffix.
+   * @param {Array} x an array
+   * @param {Array} y suffix?
+   * @param {function?} fn map function (v, i, x)
+   * @param {object?} ths this argument
+   * @returns {boolean} true if suffix
+   */
+  export function isSuffixOn<T>(x: T[], y, fn?: (v: T, i: number, x: T[]) => T[], ths?: object);
+
+  /**
+   * Gets last value.
+   * @param {Array} x an array
+   * @returns {*} last value
+   */
+  export function last<T>(x: T[]): T;
+
+  /**
+   * Returns evenly spaced values within given interval.
+   * @param {number} v start of interval
+   * @param {number} V end of interval (excluding)
+   * @param {number?} stp spacing between values (1)
+   * @returns {Array}
+   */
+  export function range<T>(v: number, V: number, stp?: number): T[];
+
+  /**
+   * Returns evenly spaced values within given interval.
+   * @param {number} v start of interval
+   * @param {number} V end of interval
+   * @param {number?} n no. of values in between (100)
+   * @returns {Array}
+   */
+  export function linspace<T>(v: number, V: number, n?: number): T[];
+
+  /**
+   * Updates values based on map function.
+   * @param {Array} x an array (updated)
+   * @param {function} fn map function (v, i, x)
+   * @param {object?} ths this argument
+   * @returns {Array} x
+   */
+  export function map$<T>(x: T[], fn?: (v: T, i: number, x: T[]) => T[], ths?: object): T[];
+
+  /**
+   * Breaks array into values, by test.
+   * @param {Array} x an array
+   * @param {function} fn test function (v, i, x)
+   * @param {object?} ths this argument
+   * @returns {Array<Array>} [[...satisfies], [...doesnt]]
+   */
+  export function partition<T>(x: T[], fn: (v: T, i?: number, x?: T[]) => boolean, ths?: object): typeof Array;
+
+  /**
+   * Breaks array into values, by map.
+   * @param {Array} x an array
+   * @param {function} fn map function (v, i, x)
+   * @param {object?} ths this argument
+   * @returns {Map<any, Array>} {key => [...values]}
+   */
+  export function partitionOn<T>(x: T[], fn?: (v: T, i: number, x: T[]) => T[], ths?: object): typeof Map;
+
+  /**
+   * Removes or replaces existing values.
+   * @param {Array} x an array
+   * @param {number} i remove index
+   * @param {number?} n no. of values to remove
+   * @param {...any} vs values to insert
+   * @returns {Array} [0->i, vs, i+n->I]
+   */
+  export function splice<T>(x: T[], i: number, n?: number, ...vs: T[]): typeof Array;
```

Comment:
This is the same as `x`:

```suggestion
  export function splice<T>(x: T[], i: number, n?: number, ...vs: T[]): T[];
```

Reply:
Noted, thanks